### DISCUSSION TITLE
How to prioritise tasks?

### DISCUSSION THREAD

**Discussion by MrMfromK** [freeplane/freeplane#847] [Help] [answered]
Hi Freeplane-team,

I am using Freeplane to organise task, ideas and errors of projects. This works very fine for me. Only one thing is not satisfying and that is prioritising the tasks. There are the round icons with numbers in it that I can add to a node. And I can filter on one of them to see what has say priority 1. But this method has some shortcomings:
- with a lot of tasks you have several ones with eg priority 1 and after filtering these appear unsorted
- changing the priority is cumbersome, remove one icon, add another
- when all tasks with priority 1 are done, I want the remaining ones to go up in priority one step, so 2 becomes 1, 3 becomes 2 and so on. Currently I have to do that manually. This is what makes this method unusable at least with many task.

I want to suggest, or ask if it already is possible, something like that:
- you can place an icon or other marking to mark a node as "included in prioritisation"
- with a menu option and / or keyboard shortcut you can open a special view showing only the nodes marked as "included in prioritisation". These are then all subnodes of one node. Their subnodes are not displayed and are not accessable as they may appear elsewhere with their own priority. Maybe there are ways to see their original parents in this view or jump back to one in the original view ( to get information about the context of this child node)
- now you can move each node up or down with the standard ways (ctrl + arrows). This changes the priority of that node and the new priority is saved somewhere in this node, so the order of priorities is remembered when next opening this special view. 
- Moving a node in the priority view does not move it in the standard view !

I hope my wish is clear, if not please ask for clarification.
I think this would be a big step for using freeplane as a project organiser.

Thanks for doing freeplane, it is a great tool.

Martin

**Comment by dpolivaev** [comment, marked as answer]
You can create a node for prioritization at any place, copy the prioritized nodes  using "copy single node" and paste them as clones (control D)

**Reply by MrMfromK** [reply]
Hi Dimitry,
thanks for the hint, I tried a little and it seems to work great. I can add a single node to the prioritisation node with ctrl-shift-C and then ctrl-D. Changes in either are reflected by the other.
I would appreciate a shortcut to jump to the original node. Currently I only found the context menu point goto. Maybe this is because one can have several clones???

**Reply by MrMfromK** [reply]
One more question:
where in the menu can I find insert as clone = ctrl-D?

**Reply by EdoFro** [reply]
> One more question: where in the menu can I find insert as clone = ctrl-D?

Edit->Paste Clone

You can use Ctrl+F1 to show the "command search" dialog. There you can type 'clone' and you will see all related menu commands and their menu position.

**Reply by MrMfromK** [reply]
Thanks, great.
I love Freeplane more every day ;-)

**Reply by EdoFro** [reply]
> I would appreciate a shortcut to jump to the original node. Currently I only found the context menu point goto. Maybe this is because one can have several clones???

I use following script to jump from one clone to the next one:

```groovy
def nTask = node

def clones = ([] + nTask.getNodesSharingContent() + nTask.getNodesSharingContentAndSubtree() + nTask).unique().sort{it.id}
//println clones*.id
//return

if(clones.size()>1) {
    def nPos = clones.indexOf(nTask)
    tgtPos = nPos < clones.size()-1 ? nPos + 1 : 0
    get(clones[tgtPos])
}

def get(n){
    n.pathToRoot*.folded = false
    c.centerOnNode(n)
    c.select(n)
}
```

I also use a node for priorization, as suggested by @dpolivaev. I call it "tasks bucket" and it has a special format style.

So, I really use 3 different scripts (all of them based in the upper script) to jump between clones:
1. one to jump to the next clone that is inside a tasks bucket
2. one to jump to the next clone that is outside a tasks bucket 
3. one to jump between only visible clones (any clone that's inside a folded branch is not considered)

This way I can clone a task in a bucket for over all priorization
 and also in a bucket where I collect the ones to do today or that I have to discuss with someone  or that I have allready started, etc.

But the script I posted here is a good starting point and You can adapt it to your needs.

hope it helps,

edo

**Comment by MrMfromK** [comment]
One more question on this:
Can I clone accross mindmaps?

**Reply by EdoFro** [reply]
No. There is no feature in Freeplane to do that.

You can simulate that feature using scripts, but I haven't seen any addon or script that currently does that.

I have made some test scripts for doing this and they work, but they are just that: test scripts. But that's why I'm sure that it can be done.

A script version, where the current map updates, each time the script runs, its nodes with the information from their changed clones in other (closed or open) maps is feasible.

What is more difficult is that the cloned nodes in different maps updates automatically each time the node has a change in any of the maps. Allthough it is posible, but both maps must be open in the same instance from Freeplane in the same computer at the time.

**Comment by euu2021** [comment]
Take a look at #548. It may be of some use for you.

Also, take at the add-ons at the [Plan and Organize](https://docs.freeplane.org/getting-started/Add-ons_(install).html#plan-and-organize) section. There are a bunch of GTD add-ons that may be useful for your usecase.
